examples/PureLogging.md

```diff
@@ -0,0 +1,157 @@
+# Pure logging
+
+One of the features that `log-warper` supports is *pure logging*.
+All related documentation and functions can be found in the [`PureLogging`][pure] module.
+In this document we will try to show how to use logging features
+in the pure functions.
+
+All the code below can be compiled and run with these commands
+
+```bash
+stack build log-warper
+stack exec pure-how-to
+```
+and you would see the logging output.
+
+## Preamble: imports and language extensions
+
+Since this is a literate haskell file, we need to specify all our language extensions
+and imports up front.
+
+``` haskell
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE OverloadedStrings          #-}
+
+module Main where
+
+
+import Control.Monad.State.Strict (MonadState (..), State, evalState, get, lift, modify)
+import Data.Semigroup ((<>))
+
+import System.Wlog (NamedPureLogger, WithLogger, WithLoggerIO, launchNamedPureLogWith,
+                    launchSimpleLogging, logDebug, logInfo)
+
+import qualified Data.Text as Text
+
+```
+
+## Pure code
+
+So, basically, if you want to log something in your function, but it doesn’t use
+the `IO` monad, you need to get some monad that implements `WithLogger`
+in your monad stack, so add the constraint and then use [`launchNamedPureLogWith`][lnplw].
+But let's have a closer look at that.
+
+Let's have this situation to give an example of how and where it should be used.
+Imagine that we have a newtype `PureCounter` which is actually wrapper on `State Int`.
+
+```haskell
+
+newtype PureCounter a = PureCounter
+    { runPureCounter :: State Int a
+    } deriving (Functor, Applicative, Monad)
+```
+
+Also we have a class 'MonadCounter' where it will all work in.
+With this class we will be able to get the current counter
+and to increase the counter by one
+
+```haskell
+
+class Monad m => MonadCounter m where
+    getCounter :: m Int
+    incCounter :: m Int
+
+```
+
+Let's implement the instance for our data type
+
+```haskell
+
+instance MonadCounter PureCounter where
+    getCounter = PureCounter $ get
+    incCounter = PureCounter $ modify (+ 1) >> get
+
+```
+
+Also some useful functions
+
+```haskell
+
+evalPureCounter :: PureCounter a -> a
+evalPureCounter pc = evalState (runPureCounter pc) 0
+
+doubleInc :: MonadCounter m => m Int
+doubleInc = incCounter >> incCounter
+
+```
+
+That `doubleInc` seems to be good candidate to add some logging, cause
+it's doing some smart stuff and it would be great to have that info logged properly.
```

Comment:
Nice! I like how you said about smart stuff :D I also think this function is too difficult and should be logged.